File: csharp/ASCrypt/RMD160.cs
using System;
using System.Security.Cryptography;

namespace ASCrypt
{
    public class RMD160
    {
        /// <summary>
        /// Computes the RIPEMD-160 checksum for the bytes.
        /// </summary>
        public static Byte[] Compute(Byte[] bytes)
        {
            RIPEMD160Managed rmd160 = new RIPEMD160Managed();
            return rmd160.ComputeHash(bytes);
        }

        /// <summary>
        /// Computes the HMAC-RIPEMD-160 for the key and bytes.
        /// </summary>
        public static Byte[] ComputeHMAC(Byte[] key, Byte[] bytes)
        {
            HMACRIPEMD160 hmac = new HMACRIPEMD160(key);
            return hmac.ComputeHash(bytes);
        }

    }

}

File: php/classes/RMD160.php
<?php

class RMD160
{
	/**
	* Computes the RIPEMD-160 checksum.
	*/	
	public static function compute($text)
	{
		return hash("ripemd160", $text, true);
	}
	
	/**
	* Computes the HMAC for RIPEMD-160.
	*/
	public static function computeHMAC($key, $text)
	{
		return hash_hmac("ripemd160", $text, $key, true);
	}
	
}

?>
